inspirehep/modules/orcid/schema.py

```diff
@@ -0,0 +1,148 @@
+
+
+@orcid_overdo.over('journal_title', 'publication_info')
+def publication_rule(self, key, value):
+    try:
+        return value[0].get('journal_title')
+    except (TypeError, KeyError):
+        pass
+
+
+@orcid_overdo.over('short_description', 'abstracts')
+def abstract_rule(self, key, value):
+    try:
+        return get_abstract({"abstracts": value})
+    except (TypeError, KeyError):
+        pass
+
+
+@orcid_overdo.over('publication_date', 'imprints')
+def date_rule(self, key, value):
+    try:
+        date = value[0].get('date')
+        date = date.split('-')
+        final_date = ['', '', '']
+        if date[0]:
+            final_date[0] = date[0]
+        if date[1]:
+            final_date[1] = date[1]
+        if date[2]:
+            final_date[2] = date[2]
+        publication_date = {'year': int(final_date[0]),
+                            'month': int(final_date[1]),
+                            'day': int(final_date[2])
+                            }
+        return publication_date
+    except (TypeError, IndexError):
+        pass
+
+
+@orcid_overdo.over('work_external_identifier', 'arxiv_eprints|dois')
+@dojson.utils.for_each_value
+def external_id_rule(self, key, value):
+    if key == 'dois':
+        return{
+            'external-identifier-type': 'DOI',
+            'external-identifier-id': value.get('value')
+        }
+    if key == 'arxiv_eprints':
+        return {
+            'external-identifier-type': 'ARXIV',
+            'external-identifier-id': value.get('value')
+        }
+
+
+@orcid_overdo.over('contributors', 'authors')
+def authors_rule(self, key, value):
+    value = utils.force_list(value)
+    orcid_authors = []
+    try:
+        orcid_authors.append({
+            'credit-name': value[0].get('full_name'),
+            'contributor-orcid': value[0].get('orcid') if 'orcid' in value[0] else '',
+            'contributor-attributes': {
+                'contributor-role': 'AUTHOR',
+                'contributor-sequence': 'FIRST'
+            }
+        })
+        for author in value[1:]:
+            orcid_authors.append({
+                'credit-name': author.get('full_name'),
+                'contributor-orcid': author.get('orcid') if 'orcid' in author else '',
+                'contributor-attributes': {
+                    'contributor-role': 'AUTHOR',
+                    'contributor-sequence': 'ADDITIONAL'
+                }
+            })
+    except KeyError:
+        pass
+    return {"contributor": orcid_authors}
+
+
+@orcid_overdo.over('type', 'collections')
+def work_type_rule(self, key, value):
+    work_types = {
+        "book": "BOOK",
+        "conferencepaper": "CONFERENCE_PAPER",
+        "proceedings": "BOOK",
+        "preprint": "WORKING_PAPER",
+        "note": "WORKING_PAPER",
+        "published": "JOURNAL_ARTICLE",
+        "thesis": "DISSERTATION",
+        "lectures": "LECTURE_SPEECH",
+        "bookchapter": "BOOK_CHAPTER",
+        "report": "REPORT"
+    }
```

Comment:
can you move this magic values in a `.config` file? Should ORCID/INSPIRE evolve on this...